docs/source/community/rfc/rfc-4.rst

```diff
@@ -0,0 +1,1013 @@
+on-disk caching of remote grids will be added. The cache will be a single
+SQLite3 database, in a user-writable directory shared by all applications using
+PROJ. Its total size will be configurable, with a default maximum size of 100 MB
+in proj.ini
+
+.. code-block:: c
+
+    /** Override, for the considered context, the path and file of the local
+    * cache of grids.
+    *
+    * @param ctx PROJ context, or NULL
+    * @param fullname Full name to the cache. If set to NULL, caching will be
+    *                 disabled.
+    * @return TRUE in case of success.
+    */
+    int proj_context_set_grid_cache_filename(PJ_CONTEXT* ctx, 
+                                             const char* fullname);
+
+    /** Override, for the considered context, the maximum size of the local
+    * cache of grids.
+    *
+    * @param ctx PROJ context, or NULL
+    * @param max_size_MB Maximum size, in mega-bytes (1024*1024 bytes)
+    * @return TRUE in case of success.
+    */
+    int proj_context_set_grid_cache_max_size(PJ_CONTEXT* ctx, int max_size_MB);
+
+The planned database structure is:
+
+.. code-block:: sql
+
+    CREATE TABLE chunks(
+        id        INTEGER PRIMARY KEY AUTOINCREMENT,
+        filename  TEXT NOT NULL,
+        offset    INTEGER NOT NULL,
+        data      BLOB NOT NULL
+    );
+
+    CREATE INDEX idx_chunks ON chunks USING (filename, offset);
+
+    CREATE TABLE recently_used(
+        age        INTEGER UNIQUE, -- 0 is last accessed
+        chunk_id   INTEGER UNIQUE NOT NULL,
+        CONSTRAINT fk_recently_used_to_chunk FOREIGN KEY (chunk_id) REFERENCES chunks(id)
+    );
+
+The chunks table will store 16 KB chunks (or less for terminating chunks).
+The recently_used table will act as a least recently used list of chunk ids.
+
+The directory used to locate this database will be ${XDG_DATA_HOME}/proj
+(per https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)
+where ${XDG_DATA_HOME} defaults to ${HOME}/.local/share on Unix builds
+and ${LOCALAPPDATA} on Windows builds. Exact details to be sorted out, but
+https://github.com/ActiveState/appdirs/blob/a54ea98feed0a7593475b94de3a359e9e1fe8fdb/appdirs.py#L45-L97
+can be a good reference.
+
+As this database might be accesse by several threads or processes at the same
+time, the code accessing to it will carefully honour SQLite3 errors regarding
+to locks, to do appropriate retries if another thread/process is currently
+locking the database. Accesses requiring a modification of the database will
+start with a BEGIN IMMEDIATE transaction so as to acquire a write lock.
+
+.. note:: This database should be hosted on a local disk, not a network one.
+          Otherwise SQLite3 locking issues are to be expected.
+
+CDN providers
++++++++++++++
+
+The following CDN providers have offered to store PROJ grids:
+
+- `Amazon Public Datasets <https://aws.amazon.com/opendata/public-datasets/>`_
+- `Fastly <https://www.fastly.com/>`_
```

Comment:
We need a section on Fastly similar to the one on Amazon below. I know that the details from Fastly are still pending, but if they fail to get back to us I think we should remove them from the list.

Reply:
we have not heard back from them yet. Maybe I'll add a " (pending confirmation and more details from the provider)" after Fastly to reflect the current status